lvsfunc/deblock.py

```diff
@@ -258,41 +252,137 @@ def _get_strength_clip(clip: vs.VideoNode, strength: SupportsFloat) -> vs.VideoN
 
-    run_dpir = DPIR(clip_rgb, strength_clip, **dpir_args)
+                continue
+
+            if isinstance(zstr, vs.VideoNode):
+                rstr_clip = zstr
+            else:
+                zstr = float(zstr)
+
+                if zstr not in cache_strength_clips:
+                    cache_strength_clips[zstr] = _get_strength_clip(clip_rgb, zstr)
+
+                rstr_clip = cache_strength_clips[zstr]
+
+            lranges = ranges if isinstance(ranges, List) else [ranges]
+
+            for rrange in lranges:
+                if rrange:
+                    dpir_zones[rrange] = rstr_clip
+
+        if len(dpir_zones) <= 2:
+            for rrange, sclip in dpir_zones.items():
+                zoned_strength_clip = replace_ranges(strength_clip, sclip, rrange)
+        else:
+            dpir_ranges_zones = {
+                range(*(
+                    (r, r + 1) if isinstance(r, int) else (r[0] or 0, r[1] + 1 if r[1] else clip.num_frames)
+                )): sclip for r, sclip in dpir_zones.items()
+            }
+
+            dpir_ranges_zones = {k: dpir_ranges_zones[k] for k in sorted(dpir_ranges_zones, key=lambda x: x.start)}
+            dpir_ranges_keys = list(dpir_ranges_zones.keys())
+
+            def _select_sclip(n: int) -> vs.VideoNode:
+                nonlocal dpir_ranges_zones, dpir_ranges_keys
+
+                for i, ranges in enumerate(dpir_ranges_keys):
+                    if n in ranges:
+                        if i > 0:
+                            dpir_ranges_keys = dpir_ranges_keys[i:] + dpir_ranges_keys[:i]
+                        return dpir_ranges_zones[ranges]
+
+                return strength_clip
+
+            zoned_strength_clip = strength_clip.std.FrameEval(_select_sclip)
+    else:
+        zoned_strength_clip = strength_clip
+
+    if backend is None:
+        if None in {cuda, fp16}:
+            try:
+                info = cast(Dict[str, int], core.trt.DeviceProperties(device_id))
+
+                fp16_available = info['major'] >= 7
+                trt_available = True
+            except BaseException:
+                fp16_available = False
+                trt_available = False
+
+        if cuda is None:
+            cuda = 'trt' if trt_available else _check_has_nvidia()
+
+        if fp16 is None:
+            fp16 = fp16_available
+
+        if cuda == 'trt':
+            channels = 2 << (not is_gray)
+
+            backend = Backend.TRT(
+                (tile_w, tile_h), fp16=fp16, num_streams=num_streams, device_id=device_id, verbose=False
+            )
+            backend._channels = channels
+        elif cuda:
+            backend = Backend.ORT_CUDA(fp16=fp16, num_streams=num_streams, device_id=device_id, verbosity=False)
+        else:
+            backend = Backend.OV_CPU(fp16=fp16)
+
+    network_path = Path(models_path) / 'dpir' / f'{tuple(DPIRModel.__members__)[model]}.onnx'
+
+    zoned_strength_clip.set_output(2)
```

Comment:
Is this set to properly cache it? If so, why not set it to something the user will likely never overlap (like idk, 200)?

Reply:
Epic smart cache workaround removed <https://github.com/Setsugennoao/lvsfunc/commit/31214d47515099adb10892883b9b70c5d42d004e>

Reply:
:clueless: